src/consumer/component-ops/components-diff.js

```diff
@@ -28,8 +29,17 @@ export default (async function componentsDiff(
   try {
     const getResults = (): Promise<DiffResults[]> => {
       if (version && toVersion) {
-        // $FlowFixMe - version, toVersion are string here, the error is unclear
-        return Promise.all(ids.map(id => getComponentDiffBetweenVersions(consumer, tmp, id, version, toVersion)));
+        // we already have the component in the components array, we just need to search for it
+        return Promise.all(
+          ids.map((id) => {
```

Comment:
Currently, it updates the ids only in the case of  `if (version && toVersion)`.
It's better to put this code above, before all the ifs.